test/utils/createPeer.js

```diff
@@ -0,0 +1,119 @@
+'use strict'
+
+/**
+ * These utilities rely on the fixtures defined in test/fixtures
+ *
+ * We create peers for use in browser/node environments
+ * configured to either connect directly (websocket listening multiaddr)
+ * or connecting through a well-known relay
+ */
+
+const Libp2p = require('libp2p')
+const multiaddr = require('multiaddr')
+const PeerId = require('peer-id')
+
+const WS = require('libp2p-websockets')
+const MPLEX = require('libp2p-mplex')
+const { NOISE } = require('libp2p-noise')
+
+const Peers = require('../fixtures/peers')
+const RelayPeer = require('../fixtures/relay')
+
+const defaultConfig = {
+  modules: {
+    transport: [WS],
+    streamMuxer: [MPLEX],
+    connEncryption: [NOISE]
+  },
+  config: {
+    pubsub: {
+      enabled: false
+    },
+    peerDiscovery: {
+      autoDial: false
+    }
+  }
+}
+
+function isBrowser() {
```

Comment:
Should we use https://github.com/ipfs/js-ipfs-utils/blob/master/src/env.js#L6 here?

Reply:
:+1: this is just my hacked attempt, will use the preferred, more robust tool

Reply:
Turns out it doesn't detect webworkers. Keeping my version for now.

Reply:
lol, I didn't read the file, they separate isBrowser from isWebworker, will add it

Reply:
:thinking:  for some reason, using it is causing problems. `isWebWorker`is not working as intended, and `isNode` may not be working? Its causing errors on the node side, using `isNode`.

I think I'd prefer to punt on this, fix in `libp2p-test`

Reply:
wow, just doing `!isNode` does not get that working?

Reply:
if it does not work, we can punt yes. But that is strange because we are using that across all IPFS land

Reply:
it works, in some respect, but causes tests in node fail, with unhandled promise rejection, very strange. Specifically the floodsub.spec.js 'subscription functionality' 'after all' hook. Didn't look into it much more than that.

Reply:
ok, lets leave it for now